### DISCUSSION TITLE
предложения - BreackGraph.OnDamage и BreackGraph.OnBroke

### DISCUSSION THREAD

**Discussion by justqstn** [Sokol-Games-GmbH/PixelCombats.PublicApi#42] [Ideas]
ApiEvent<index Index, ushort BlockId, ApiPlayer player<> OnDamage; // когда блок был поврежден кем-то или игроком
ApiEvent<index Index, ushort BlockId, ApiPlayer player> OnBroke; // когда блок был сломан

BreackGraph.OnDamage.Add(function(pos, id, plr) {
    if (pos.x == 1 && pos.y == 1 && pos.z == 1 && id == 1) plr.Properties.Scores.Value += 10;
    // когда игрок повреждает блок с id 1 на координатах "1,1,1", он получает 10 очков 
});

BreackGraph.OnBroke.Add(function(pos, id, plr) {
    if (pos.x == 1 && pos.y == 1 && pos.z == 1 && id == 1) plr.Properties.Scores.Value += 1000;
    // когда игрок ломает блок с id 1 на координатах "1,1,1", он получает 1000 очков 
});

**Comment by kkohno** [comment]
там еще и причина поломки у нас есть, если что... тоже можно добавить, хорошее предложение
